compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -206,17 +202,7 @@ impl<'tcx> Inliner<'tcx> {
             return Ok(());
         }
 
-        if let Some(callee_def_id) = callee.def_id().as_local() {
-            let callee_hir_id = self.tcx.hir().local_def_id_to_hir_id(callee_def_id);
-            // Avoid a cycle here by only using `instance_mir` only if we have
-            // a lower `HirId` than the callee. This ensures that the callee will
-            // not inline us. This trick only works without incremental compilation.
-            // So don't do it if that is enabled.
-            if !self.tcx.dep_graph.is_fully_enabled() && self.hir_id.index() < callee_hir_id.index()
-            {
-                return Ok(());
-            }
-
+        if callee.def_id().as_local().is_some() {
```

Comment:
Marking as blocked on https://github.com/rust-lang/rust/pull/85321.

Reply:
https://github.com/rust-lang/rust/pull/85321 is now merged. Removing S-blocked.